src/web/wms/src/main/java/org/geoserver/wms/web/data/AbstractStylePage.java

```diff
@@ -430,7 +432,59 @@ public void setRawStyle(Reader in) throws IOException {
         editor.setModelObject(rawStyle);
         in.close();
     }
-    
+
+    /**
+     * Check for an original CSS version of the style created by the old CSS extension (pre-pluggable styles). If a CSS style is found, recover it if
+     * the derived SLD has not subsequently been manually edited.
+     * 
+     * The recovery is accomplished by updating the catalog to point to the original CSS file, and changing the style's format to "css".
+     * 
+     * @param si The {@link StyleInfo} for which to check for and potentially recover a CSS version.
+     */
+    protected void recoverCssStyle(StyleInfo si) {
+        if (si == null) {
+            return;
+        }
+
+        // Only try to repair missing CSS files if a CSS style handler is registered.
+        try {
+            Styles.handler("css");
+        } catch (Exception e) {
+            return;
+        }
+
+        // Use this tolerance to prevent erasing an SLD that was manually edited after being
+        // generated from a CSS file. (Generated SLDs will always be newer than the CSS).
+        long favorSLDIfNewerByMS = 600000;
+
+        // The problem only exists for styles that were not explicitly set as "css".
+        if (!"css".equalsIgnoreCase(si.getFormat())) {
```

Comment:
Yep, plus stuff like YSLD wasn't added until much more recently, so we don't want to be accidentally clobbering it.